SonarQube.TeamBuild.PreProcessor/AnalysisConfigGenerator.cs

```diff
@@ -72,9 +81,12 @@ public static class AnalysisConfigGenerator
                 }
             }
 
-            // Add command line arguments
+            // Add command line arguments and programmatic properties.
+            // Command line arguments take precedence
+            AggregatePropertiesProvider aggProvider = new AggregatePropertiesProvider(args.LocalProperties, new ListPropertiesProvider(programmaticProperties));
```

Comment:
The _AggregatePropertiesProvider_ already handles the details of property precedence for us, so we can just use it here to ensure than any command line properties "win" over programmatic properties of the same name.

A side-effect of this is that the user will be able to specify the build wrapper output location from the command line if they need to.